### ISSUE TITLE
XQuery: index rewritings on collections

### ISSUE THREAD

**Issue by ChristianGruen** [BaseXdb/basex#1067]
The query in the following command script uses no index structure:

```xquery
set autooptimize on
create db collection
add to 1.xml <X>A</X>
add to 2.xml <X>B</X>
xquery collection("collection")[.//text() = 'A']
```

**Comment by ChristianGruen** [comment]
Resulting query plan:

```xml
<QueryPlan compiled="true">
  <IterFilter>
    <DBNodeSeq size="2">
      <DBNode name="collection" pre="0"/>
      <DBNode name="collection" pre="3"/>
    </DBNodeSeq>
    <CmpG op="=">
      <CachedPath>
        <IterStep axis="descendant" test="text()"/>
      </CachedPath>
      <Str value="A" type="xs:string"/>
    </CmpG>
  </IterFilter>
</QueryPlan>
```

Pattern to look for: filter expression with nodes (DBNodeSeq) and index-rewritable expression (`CmpG`, etc.) in predicate. Basically the same as in path expressions.

**Comment by ChristianGruen** [comment]
Resolved: 023910046b0a120f44bdccdc8ea9ae1bc1b2d198
